src/docfx/build/toc/ServicePageGenerator.cs

```diff
@@ -86,9 +86,10 @@ internal class ServicePageGenerator
                                 hrefFileFullPath = Path.GetFullPath(Path.Combine(topLevelTOCYmlDir == null ? "" : topLevelTOCYmlDir, childHref));
                             }
 
-                            var servicePageFullPath = Path.GetDirectoryName(Path.GetFullPath(Path.Combine(_docsetPath, servicePagePath.Path))) ?? _docsetPath;
-                            var hrefRelativePath = Path.GetRelativePath(servicePageFullPath, hrefFileFullPath);
-                            childHref = hrefRelativePath;
+                            var hrefRelativePathToDocset = Path.GetRelativePath(_docsetPath, hrefFileFullPath);
+                            var hrefFullPathOfDocset = $"~/{hrefRelativePathToDocset}";
```

Comment:
`~` does not always represent to `_docsetPath`

Reply:
Fixed.